qualtran/resource_counting/_bloq_counts.py

```diff
@@ -128,21 +135,69 @@ class GateCounts:
     cswap: SymbolicInt = 0
     and_bloq: SymbolicInt = 0
     clifford: SymbolicInt = 0
-    rotation: SymbolicInt = 0
     measurement: SymbolicInt = 0
+    binned_rotation_epsilons: Counter[int] = field(
+        factory=Counter, converter=_mapping_to_counter, eq=lambda d: tuple(d.items())
+    )
+    eps_bin_prec: int = 20
```

Comment:
please update the docstring with a thorough description of the representation of rotations. 

Is it possible for someone to construct this class with inconsistent binned values vs the `eps_bin_prec`? What if someone puts in the epsilons as floats instead of 2**20*floats 

document that the `eps_bin_prec` behavior during addition. 

`eps_bin_prec` -> `epsilon_precision`? and then document that precision is measured by number of binary digits.

Reply:
Used `np.format_scientific_float` to avoid such issues.